csrc/alias_analysis.cpp

```diff
@@ -59,6 +59,14 @@ bool okToRelayout(
     const TensorView* tv,
     const Layout& new_layout,
     const EmptyAllocationAs empty_allocation_as) {
+  // we can merge this with the one below
+  // when using logical domain as the allocation domain, we can ignore the
+  // layout when it's not used by codegen
+  if (empty_allocation_as == EmptyAllocationAs::kLogical &&
+      !ir_utils::canUsePresetAllocationDomain(tv, false)) {
```

Comment:
Is it possible to separate this into another PR? It's unclear to me why this PR needs this, and the adding-a-knob-to-choose-between-two-code-paths approach can create troubles for maintenance.